psydac/api/ast/fem.py

```diff
@@ -767,8 +766,10 @@ def _create_ast_bilinear_form(terminal_expr, atomic_expr_field,
             es             = dict()
             for v in sub_tests:
                 v_str = str(SymbolicExpr(v))
-                bs[v] = variables(('b_{v}_1, b_{v}_2, b_{v}_3'.format(v=v_str)), dtype='int')[:dim] if is_parallel else [S.Zero]*dim
-                es[v] = variables(('e_{v}_1, e_{v}_2, e_{v}_3'.format(v=v_str)), dtype='int')[:dim] if is_parallel else [S.Zero]*dim
+#                bs[v] = variables(('b_{v}_1, b_{v}_2, b_{v}_3'.format(v=v_str)), dtype='int')[:dim] if is_parallel else [S.Zero]*dim
+#                es[v] = variables(('e_{v}_1, e_{v}_2, e_{v}_3'.format(v=v_str)), dtype='int')[:dim] if is_parallel else [S.Zero]*dim
+                bs[v] = [S.Zero]*dim
+                es[v] = [S.Zero]*dim
```

Comment:
What is contained in the dictionaries `bs` and `es`?

Reply:
Before when we had the ghost elements, we added an optimization to calculate only what we needed, so `bs` and `es` contains the `starts` and the `ends` of the test functions loops.
Now with the new changes, we don't need that anymore, but I didn't delete it because I was thinking of the case where the communications take the same or more time than the calculations, in that case, we can disable them and put back the ghost elements, this is needed when the degrees are small like `1` or `2` and the number of processes is really big which makes the elements owned by a process really small and the calculations really fast.
This optimization can be added after improving the communications.

Reply:
Thanks for the explanation! Can you please copy & paste it as a big comment in the code?

Reply:
done in [e64962d](https://github.com/pyccel/psydac/pull/232/commits/e64962dbeccdf5b5529742544f3327b3f2ba2410)